File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = []
    
    for i in s:
        if i == "(":
            answer.append(i)
        elif i == ")":
            if len(answer) == 0:
                return False 
            else:
                answer.pop()
                
    if answer:
        return False
    else:
        return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    
    let arr = [];
    
    for(let value of s){
        if(value === ")" && arr.length === 0) {
            arr.push(value);
        }
        else if(value === "("){
            arr.push(value);
        }
        else if(value === ")" && arr[arr.length - 1] === "("){
            arr.pop();
        }
    }
    
    return arr.length === 0 ? true : false;
}